_posts/2024/2024-02-06-about-caching-in-eo.md

```diff
@@ -0,0 +1,241 @@
+2) Then `Gradle` executes the task and saves a fingerprint of the path and contents of the output files.
+3) Then, when Gradle starts a project build again, it generates a new fingerprint for the same files.
+   If the new fingerprint has not changed, Gradle can safely skip this task. 
+   In the opposite case, the task needs to perform an action and to rewrite outputs.
+   The fingerprint is considered current if the last modification time 
+   and the size of the source files have not changed.
+
+
+In addition to `Incremental build`, `Gradle` also stores fingerprints of previous builds, enabling quick project builds,
+for example when switching from one branch to another. This feature is known as 
+the [Build Cache](https://docs.gradle.org/current/userguide/build_cache.html).
+
+
+### Maven
+[Maven](https://maven.apache.org) automates and manages Java-project builds.
+`Maven` is based on the concept of 
+[Maven LifeCycles](https://maven.apache.org/guides/introduction/introduction-to-the-lifecycle.html), 
+which include default, clean, and site lifecycles. 
+Each lifecycle consists of `phases` and these `phases` consist of sets of `goals`.
+
+In Maven, there are default phases and goals for building any projects:
+
+<p align="center">
+  <img src="/images/defaultPhaseMaven.svg">
+</p>
+
+By default, the `phases` in Maven are inherently connected within the build lifecycle.
+Each `phase` represents a specific task, and the execution order of `goals` within `phases` is determined 
+by the default Maven lifecycle bindings. This means that while each `phase` operates as a series of individual tasks
+and their execution order is predefined by Maven.
+
+
+`Maven` utilizes caching mechanisms through the `takari-lifecycle-plugin` and `maven-build-cache-extension`:
+* The [takari-lifecycle-plugin](http://takari.io/book/40-lifecycle.html) is an alternative to the default Maven lifecycle 
+(building JAR files). Its distinctive feature is the use of a single universal plugin with the same functionality 
+as five separate plugins for the standard lifecycle, but with significantly fewer dependencies. As a result,
+it provides a much faster startup, more optimal operation, and lower resource consumption.
+This leads to a significant increase in performance when compiling complex projects with a large number of modules.
+
+* The [maven-build-cache-extension](https://maven.apache.org/extensions/maven-build-cache-extension/)
+is used for large Maven projects that have a significant number of small `modules`.
+A `module` refers to a subproject within a larger project.
+Each `module` has its own `pom.xm` file, and there is an aggregator `pom.xml` that consolidates all the `modules`.
+This plugin takes a key for a `module`, it encapsulates the essential aspects of the `module`,
+including the source code and the configuration of the plugins used within it.
+`Modules` with the same key are current or unchanged and the cache can efficiently restore them.
```

Comment:
@Yanich96 This sentence doesn't make sense to me: "`Modules` with the same key are current or unchanged and the cache can efficiently restore them.".
By the way, what is the "key"?

Reply:
@volodya-lombrozo key is hash. Can you explain me your point: "This sentence doesn't make sense to me"? Do you mean this is obvious?

Reply:
@Yanich96  I don't understand "current" word here